fuzz/quicly_mock.c

```diff
@@ -0,0 +1,457 @@
+/*
+ * Copyright (c) 2021 Fastly, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include <assert.h>
+#include <malloc.h>
+#include <stdbool.h>
+#include "khash.h"
+#include "quicly.h"
+#include "quicly/sendstate.h"
+#include "quicly/recvstate.h"
+#include "quicly_mock.h"
+
+KHASH_MAP_INIT_INT64(quicly_stream_t, quicly_stream_t *)
+
+mquicly_context_t mquicly_context;
+
+struct st_quicly_conn_t {
+    struct _st_quicly_conn_public_t super;
+    khash_t(quicly_stream_t) * streams;
+};
+
+struct st_quicly_send_context_t {
+};
```

Comment:
Do we use this?

Reply:
Not really, but `quicly_send_t` takes `quicly_send_context_t`, which is defined as a typedef of `struct st_quicly_sed_context_t` in quicly.h. So we have to provide a structure type `struct st_quicly_sed_context_t`.

Reply:
Thank you for the explanation. Makes sense.